### PR TITLE
Sidebar toggle enabled

### LINKED ISSUE #29: Menu button not working
Theme Reference: https://gogo-react.coloredstrategies.com/app/dashboards/default
Capture
In mobile view: Menu button not working

### PR DESCRIPTION
**Pull request by coderrsid**
# FIXES #29 

Screen-Recording-2020-05-17-at-1

### CODE REVIEW

File: app/layouts/Dashboard/index.js [hunk 1]
```diff
@@ -1,16 +1,21 @@
...
+
+  const toggleSideBar = () => {
+    setSideBar(!showSideBar);
+  };
 
   return (
     <div className="dashboard-layout">
-      <Navbar />
-      <Sidebar />
+      <Navbar handleSideBar={toggleSideBar} />
```
Comment (0xnmn) [inline app/layouts/Dashboard/index.js]: rename it to `toggleSideBar`

File: app/layouts/Dashboard/index.js [hunk 2]
```diff
@@ -1,16 +1,21 @@
...
+      {showSideBar && <Sidebar />}
```
Comment (0xnmn) [inline app/layouts/Dashboard/index.js]: Use classnames to hide the sidebar and add animations. Refer to this: https://gogo-react.coloredstrategies.com/app/dashboards/default

File: app/components/Navbar/index.js [hunk 3]
```diff
@@ -13,15 +13,17 @@ import useViewer from 'Hooks/useViewer';
...
   const viewer = useViewer();
   const [showUserMenu, setUserMenu] = useState(false);
 
   return (
     <nav className="navbar fixed-top p-none">
       <div className="row w-100">
         <div className="col-4">
-          <div className="menu-button pointer d-md-block text-left">
+          <div
+            onClick={props.handleSideBar}
```
Comment (0xnmn) [inline app/components/Navbar/index.js]: rename this

File: app/components/Navbar/index.js [hunk 4]
```diff
@@ -13,15 +13,17 @@ import useViewer from 'Hooks/useViewer';
 
 import './style.scss';
 
-const Navbar = () => {
+const Navbar = (props) => {
```
Comment (0xnmn) [inline app/components/Navbar/index.js:16]: destruct and extract the used props.

**Review by 0xnmn — CHANGES_REQUESTED** [review]

File: app/layouts/Dashboard/index.js (hunk 2 above)
Comment (coderrsid) [reply app/layouts/Dashboard/index.js]: so you mean to wrap up in a div and use classes to toggle right?

Comment (coderrsid) [reply app/layouts/Dashboard/index.js]: or to pass props to sidebar and using them toggle the sidebar component itself!

Comment (0xnmn) [reply app/layouts/Dashboard/index.js]: pass state to the sidebar and let sidebar add or remove appropriate classes to the div it is rendering.

Comment (coderrsid) [reply app/layouts/Dashboard/index.js]: Sure. I'll just add the functionality.

Comment (coderrsid) [reply app/layouts/Dashboard/index.js:18]: image

How can i receive props here withRouter?

Comment (coderrsid) [reply app/layouts/Dashboard/index.js]: I tried to receive props but couldn't correctly i gues..
image

### pressed 6 times
image

Comment (0xnmn) [reply app/layouts/Dashboard/index.js]: I'll let you spend 2-3 more hours figuring out what is wrong. That's the fun of it.

File: app/layouts/Dashboard/index.js (hunk 1 above)
Comment (coderrsid) [reply app/layouts/Dashboard/index.js]: Done!

File: app/layouts/Dashboard/index.js (hunk 2 above)
Comment (coderrsid) [reply app/layouts/Dashboard/index.js]: Done!

File: app/components/Navbar/index.js (hunk 3 above)
Comment (coderrsid) [reply app/components/Navbar/index.js]: Done!

File: app/components/Navbar/index.js (hunk 4 above)
Comment (coderrsid) [reply app/components/Navbar/index.js:16]: Done!

**Comment by 0xnmn** [comment]
@coderrsid The issue reaming to solve is, the main content doesn't shift to left when the sidebar is closed. Also on mobile devices by default the sidebar is open which we don't want.

Please also add the latest gif showing the animations.

**Comment by coderrsid** [comment]
> @coderrsid The issue reaming to solve is, the main content doesn't shift to left when the sidebar is closed. Also on mobile devices by default the sidebar is open which we don't want.

So, i tried to see how this is implemented in Gogo React, there the margins of `<main>` div in Dashboard has been done through a media query, so should i add that too? 
image

> Please also add the latest gif showing the animations.

Yeah sure. 
Screen-Recording-2020-05-18-at-1

**Comment by 0xnmn** [comment]
> @coderrsid The issue reaming to solve is, the main content doesn't shift to left when the sidebar is closed. Also on mobile devices by default the sidebar is open which we don't want.
> 
> Please also add the latest gif showing the animations.

This ain't fixed. When on large screens displays the main content should move the left when the sidebar is closed.

**Comment by coderrsid** [comment]
> This ain't fixed. When on large screens displays the main content should move the left when the sidebar is closed.

So can i just wrap sidebar and main and add respective classes .. or any other method.. coz i want to do it the right way.

**Comment by 0xnmn** [comment]
> > This ain't fixed. When on large screens displays the main content should move the left when the sidebar is closed.
> 
> So can i just wrap sidebar and main and add respective classes .. or any other method.. coz i want to do it the right way.

Sidebar and main are in same components, you just need to set appropriate class on main to fix it.

**Comment by coderrsid** [comment]
> Sidebar and main are in same components, you just need to set appropriate class on main to fix it.

Actually, it was the outer div for which classes needed to be added. Done though. Please review @hereisnaman !

Screen-Recording-2020-05-19-at-1

**Comment by 0xnmn** [comment]
@coderrsid perfection. Resolve the conflicts and I'll merge.

**Comment by coderrsid** [comment]
> @coderrsid perfection. Resolve the conflicts and I'll merge.

Thanks 😄 Resolved the conflicts!

**Review by 0xnmn — APPROVED** [review]

**Pull request merged**